### PR TITLE
Add a lecture on optimal investment

### PR DESCRIPTION
**Pull request by Smit-create**
#9

### CODE REVIEW

File: lectures/opt_invest.md [hunk 1]
```diff
@@ -0,0 +1,610 @@
...
+```
+
++++ {"user_expressions": []}
+
+Similarly, let's define the matrix version to compute $\sigma$-policy operator.
+
+```{code-cell} ipython3
+def T_σ_matrix_version(v, σ, constants, sizes, arrays):
+    "The σ-policy operator, single index version."
+
```
Comment (Smit-create) [inline lectures/opt_invest.md:434]: @jstac We don't actually use these functions and I guess they were just for verifying. Should we remove them? as we already have other faster function computing the same.

**Comment by jstac** [comment]
Hi @Smit-create , is this ready for me to review?

**Comment by jstac** [comment]
Sorry @Smit-create , ignore that, I see your review request, will do so now.

**Comment by jstac** [comment]
@Smit-create I looked this over and decided that you were right: we should get rid of the matrix versions.

In which case, I think we can also get rid of `def compute_P_σ(σ, constants, sizes, arrays):` ?

Having two versions is too confusing.

Please delete those and then ping me.

@chappiewuzefan You might find that the optimal savings problem in #10 is similar.  Please consult with @Smit-create if you are not sure.

**Comment by Smit-create** [comment]
Hi @jstac, I have removed the matrix versions.

**Comment by jstac** [comment]
This is perfect, thanks @Smit-create !

**Pull request merged**